docs/reference/hooks.md

```diff
@@ -102,6 +102,28 @@ depth: 1
 ---
 ```
 
+## Appearance
+
+Hooks for modifying the display and appearance of basic CMS features and furniture.
+
+(get_avatar_url)=
+
+### `get_avatar_url`
+
+There is an existing `avatar` field on the Wagtail `UserProfile` model, which, although it cannot be altered on the model level, can be intercepted at the view level and made to point to some other image before it is rendered. Just define the hook and resolve the value to whatever image url you want to see taking the place of the `UserProfile` (or gravatar) image.
+
+For example, you might have an avatar on a `Profile` model in your own application that is keyed to the `auth.User` model in the familiar way. In that case, you could register your hook as:
+
+```python
+@hooks.register('get_avatar_url')
+def get_profile_avatar(user, size):
+    return user.profile.avatar
+```
+
+And the `UserProfile` avatar will be replaced with your own `Profile` avatar accordingly.
+
+Additionally, you can use the default `size` parameter that is passed in to the hook if you need to do any further processing before retrieving your url.
```

Comment:
This is a bit too much like a how-to guide rather than reference documentation (see https://docs.wagtail.org/en/stable/contributing/documentation_modes.html for some background) - I think it would be best to follow the same kind of wording as the other hooks ("The callable passed to this hook receives the arguments ... and should return ...").

It would also be good to have a standalone code example here that demonstrates the usage without having to introduce the idea of having alternative user profile models. Perhaps it could be a function that returns `/static/images/santa.png` if  today's date is December 25th - something easily understandable and not too contrived...

Then, if you feel it's still worth documenting the "real world" use case of an image on an alternative profile model, that would be a good fit for the customization/admin_templates.md page.